tests/publiccloud/azure_nfs.pm

```diff
@@ -0,0 +1,84 @@
+# SUSE's openQA tests
+#
+# Copyright 2022 SUSE LLC
+# SPDX-License-Identifier: FSFAP
+
+# Summary: Test Azure NFS File shares
+# Maintainer: qa-c team <qa-c@suse.de>
+
+use base 'consoletest';
+use strict;
+use warnings;
+use testapi;
+use utils;
+use publiccloud::utils "select_host_console";
+
+sub check_nfs_share {
+    my $share = shift;
+
+    ##  NFS share test: Download some data and check permissions
+    # * Download a file, and set the permission to read-only by owner
+    # * Check if the permissiosn are correctly set
+    # * Check if the default user (azureuser) can not read the file
+    # * Check if hard links are working
+    # * Check if soft links are working
+    assert_script_run("cd $share");
+    assert_script_run("curl -v -o Big_Buck_Bunny_8_seconds_bird_clip.ogv " . data_url('Big_Buck_Bunny_8_seconds_bird_clip.ogv'));
+    assert_script_run("sync");
+    assert_script_run("chmod 0400 Big_Buck_Bunny_8_seconds_bird_clip.ogv");
+    assert_script_run("chown root:root Big_Buck_Bunny_8_seconds_bird_clip.ogv");
+    validate_script_output("ls -al Big_Buck_Bunny_8_seconds_bird_clip.ogv", sub { m/^-r--------.*/ });
```

Comment:
`stat -c "%a" Big_Buck_Bunny_8_seconds_bird_clip.ogv` should return permissions in octal format. That might be considered as more readable